mllib/src/main/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifier.scala

```diff
@@ -369,11 +377,18 @@ object MultilayerPerceptronClassificationModel
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
 
       val dataPath = new Path(path, "data").toString
-      val data = sparkSession.read.parquet(dataPath).select("layers", "weights").head()
-      val layers = data.getAs[Seq[Int]](0).toArray
-      val weights = data.getAs[Vector](1)
-      val model = new MultilayerPerceptronClassificationModel(metadata.uid, layers, weights)
-
+      val columns = sparkSession.read.parquet(dataPath).columns
+      val model = if (columns.length == 2) { // model prior to 3.0.0
+        val data = sparkSession.read.parquet(dataPath).select("layers", "weights").head()
+        val layers = data.getAs[Seq[Int]](0).toArray
+        val weights = data.getAs[Vector](1)
+        val model = new MultilayerPerceptronClassificationModel(metadata.uid, weights)
+         model.setLayers(layers)
+      } else {
+        val data = sparkSession.read.parquet(dataPath).select("weights").head()
+        val weights = data.getAs[Vector](0)
+        new MultilayerPerceptronClassificationModel(metadata.uid, weights)
```

Comment:
Pardon the dumb question, but, why does the new model version not need to load layers? is it part of weights?

Reply:
In the new code, since ```MultilayerPerceptronClassificationModel``` extends ```MultilayerPerceptronParams```, ```layers``` is one of the Params, and it is set in the next line:
```
metadata.getAndSetParams(model)
```